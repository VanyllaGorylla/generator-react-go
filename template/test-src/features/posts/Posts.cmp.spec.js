import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import moxios from 'moxios';

import WrapperPostsCmp, { PostsCmp } from '@/features/posts/Posts.cmp';

describe('Posts.cmp', function() {
  let store;

  beforeEach(() => {
    moxios.install();
    store = configureStore()({
      posts: {
        data: [
          {
            id: 1,
            title: 'First Post',
            content: 'Posts content'
          },
          {
            id: 2,
            title: 'Second Post',
            content: 'Posts content'
          },
          {
            id: 3,
            title: 'Third Post',
            content: 'Posts content'
          }
        ],
        dataLoading: false,
        crudLoading: false
      }
    });
  });

  it('should render proper HTML when data is provided and there is no loading atm', function() {
    moxios.stubRequest('/posts', {
      status: 200
    });

    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <WrapperPostsCmp />
          </Router>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render proper HTML when loading is performing atm', function() {
    const tree = renderer
      .create(
        <Router>
          <PostsCmp data={[]} dataLoading={true} />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(function() {
    moxios.uninstall();
  });
});
