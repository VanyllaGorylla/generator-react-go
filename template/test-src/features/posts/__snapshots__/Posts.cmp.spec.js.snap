// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts.cmp should render proper HTML when data is provided and there is no loading atm 1`] = `
<div
  className="container"
>
  <div
    className="card"
  >
    <div
      className="card-content"
    >
      <span
        className="card-title grey-text text-darken-4"
      >
        Posts
      </span>
      <div>
        <div
          className="row"
        >
          <div
            className="col s12 m12 l12"
          >
            <ul
              className="collection"
            >
              <li
                className="collection-item"
              >
                First Post
                <div
                  className="right"
                >
                  <span
                    className="action-element action-element__delete"
                    onClick={[Function]}
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-trash"
                      data-id={1}
                    />
                  </span>
                  <span
                    className="action-element "
                  >
                    <a
                      href="/posts/edit/1"
                      onClick={[Function]}
                    >
                      <i
                        aria-hidden="true"
                        className="fa fa-pencil "
                      />
                    </a>
                  </span>
                </div>
              </li>
              <li
                className="collection-item"
              >
                Second Post
                <div
                  className="right"
                >
                  <span
                    className="action-element action-element__delete"
                    onClick={[Function]}
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-trash"
                      data-id={2}
                    />
                  </span>
                  <span
                    className="action-element "
                  >
                    <a
                      href="/posts/edit/2"
                      onClick={[Function]}
                    >
                      <i
                        aria-hidden="true"
                        className="fa fa-pencil "
                      />
                    </a>
                  </span>
                </div>
              </li>
              <li
                className="collection-item"
              >
                Third Post
                <div
                  className="right"
                >
                  <span
                    className="action-element action-element__delete"
                    onClick={[Function]}
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-trash"
                      data-id={3}
                    />
                  </span>
                  <span
                    className="action-element "
                  >
                    <a
                      href="/posts/edit/3"
                      onClick={[Function]}
                    >
                      <i
                        aria-hidden="true"
                        className="fa fa-pencil "
                      />
                    </a>
                  </span>
                </div>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
    <div
      className="card-reveal"
    >
      <span
        className="card-title grey-text text-darken-4"
      >
        Posts
        <i
          className="material-icons right"
        >
          close
        </i>
      </span>
    </div>
    <div
      className="card-action"
    >
      <a
        href="/posts/new"
        onClick={[Function]}
      >
        Create new
      </a>
    </div>
  </div>
</div>
`;

exports[`Posts.cmp should render proper HTML when loading is performing atm 1`] = `
<div
  className="container"
>
  <div
    className="card"
  >
    <div
      className="card-content"
    >
      <span
        className="card-title grey-text text-darken-4"
      >
        Posts
      </span>
      <div>
        <div
          className="row"
        >
          <div
            className="col s12 m12 l12"
          >
            <div
              className="progress"
            >
              <div
                className="indeterminate"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="card-reveal"
    >
      <span
        className="card-title grey-text text-darken-4"
      >
        Posts
        <i
          className="material-icons right"
        >
          close
        </i>
      </span>
    </div>
    <div
      className="card-action"
    >
      <a
        href="/posts/new"
        onClick={[Function]}
      >
        Create new
      </a>
    </div>
  </div>
</div>
`;
